Tests/SonarScanner.MSBuild.PreProcessor.Test/PreprocessorObjectFactoryTests.cs

```diff
@@ -87,41 +87,76 @@ public async Task CreateSonarWebService_MissingUriScheme_ReturnNullAndLogErrors(
     }
 
     [DataTestMethod]
-    [DataRow("8.0", typeof(SonarCloudWebServer))]
-    [DataRow("9.9", typeof(SonarQubeWebServer))]
-    public async Task CreateSonarWebServer_CorrectServiceType(string version, Type serviceType)
+    [DataRow("https://sonarcloud.io", "8.0", typeof(SonarCloudWebServer))]
+    [DataRow("https://sonarcloud.io/", "8.0", typeof(SonarCloudWebServer))]
+    [DataRow("https://sonarcloud.io//", "8.0", typeof(SonarCloudWebServer))]
+    [DataRow("https://sonarcloud_other.io//", "8.1", typeof(SonarQubeWebServer))]
+    [DataRow("http://localhost:222", "8.9", typeof(SonarQubeWebServer))]
+    public async Task CreateSonarWebServer_CorrectServiceType(string hostUrl, string version, Type serviceType)
     {
         var sut = new PreprocessorObjectFactory(logger);
         var downloader = Substitute.For<IDownloader>();
         downloader.Download(Arg.Any<string>(), Arg.Any<bool>()).Returns(Task.FromResult(version));
 
-        var service = await sut.CreateSonarWebServer(CreateValidArguments(), downloader);
+        var service = await sut.CreateSonarWebServer(CreateValidArguments(hostUrl), downloader);
 
         service.Should().BeOfType(serviceType);
     }
 
-    [TestMethod]
-    public async Task CreateSonarWebServer_ValidCallSequence_ValidObjectReturned()
+    [DataTestMethod]
+    [DataRow("https://sonarcloud.io", "8.9", true)]
+    [DataRow("https://sonarqube.gr", "8.0", false)]
+    [DataRow("http://localhost:4242", "8.0", false)]
```

Comment:
Here what is happening is that the user sets something but the server is different.
For example, on the first case the user sets `sonarcloud io`, but the server is actually `SQ 8.9`.
On the second one it is the opposite, the user sets sonarqube, but the server is actually SC (found by quering the version 8.0)